Sources/InstantSearchCore/Searcher/Hits/HitsSearcher.swift

```diff
@@ -147,34 +149,40 @@ public final class HitsSearcher: IndexSearcher<AlgoliaSearchService> {
     - Parameters:
        - index: Index value in which search will be performed
        - query: Instance of Query. By default a new empty instant of Query will be created.
+       - isAutoSendingHitsViewEvents: flag defining whether the automatic hits view Insights events sending is enabled
        - requestOptions: Custom request options. Default is nil.
    */
   public init(client: SearchClient,
               indexName: IndexName,
               query: Query = .init(),
+              isAutoSendingHitsViewEvents: Bool = false,
               requestOptions: RequestOptions? = nil) {
     let service = AlgoliaSearchService(client: client)
     let request = AlgoliaSearchService.Request(indexName: indexName, query: query, requestOptions: requestOptions)
     super.init(service: service, initialRequest: request)
     Telemetry.shared.trace(type: .hitsSearcher,
                            parameters: .client)
     onResults.subscribe(with: self) { searcher, response in
-      searcher.eventTracker.trackView(for: response.hits,
-                                     eventName: "Hits Viewed")
+        if isAutoSendingHitsViewEvents {
+            searcher.eventTracker.trackView(for: response.hits, eventName: "Hits Viewed")
+        }
```

Comment:
Could we rather send the view events when the hits enter the viewport as batches?

I'm afraid this PR solution creates more noise, especially for Personalization, which is our only product relying on search view events AFAIK. It blurs the signal because there's no guarantee that the user actually sees all the hits from the response.

I created a [more correct approach](https://github.com/algolia/ai-personalization-demo/blob/e53971a8ec4da0e5c09494fdec837e91ee96551c/app/lib/useHitsObserver.ts) in JavaScript for the Personalization Storefront, perhaps the principles could be reused?

Reply:
Events are actually sent as batches under the hood. We are using [InstantSearch Insights](https://github.com/algolia/instantsearch-ios/tree/master/Sources/InstantSearchInsights) which handles both batching and the offline situation.

Reply:
Yeah but this doesn't address the lack of viewport issue. All hits are considered viewed when results are received, which isn't even close to reality.